### DISCUSSION TITLE
Passing custom param to IdP app

### DISCUSSION THREAD

**Discussion by glebsts** [keycloak/keycloak#35699] [Q&A]
Hello,
I have few client applications configured to use same realm in Keycloak, and single Identity provider application used for this realm.
1) I'd like to get stats from IdentityProvider about e.g. how many auth session happened for each client
2) One of client apps now got ui language selection, identity provider app also supports multiple languages. I'd like IdP app to display UI in language user previously selected in client app.

First thing coming to my mind is passing some parameters to IdP application. 
For case (1) I could imagine passing client_id in provider Authorization URL. 
For case (2) I could imagine client app passing selected_language parameter to Keycloak URL, and Keycloak would pass it further in same provider Authorization URL.

Obviously I was not able to find straightforward way to configure it. 
Any chance there is a proper way to do both in Keycloak? v22.0.0.1 if it matters.

**Comment by RalfsVetra** [comment]
Go to Admin Console -> Identity Providers -> Your Provider -> Forwarded query parameters.

When your app redirects to the keycloak, specify the required parameters in the url.
